daemon/health_http_linux.go

```diff
@@ -0,0 +1,41 @@
+// +build linux
+
+package daemon
+
+import (
+	"golang.org/x/net/context"
+
+	"github.com/docker/docker/container"
+	"github.com/docker/engine-api/types"
+	"github.com/docker/engine-api/types/strslice"
+	"github.com/vishvananda/netns"
+)
+
+func (p *httpProbe) run(ctx context.Context, d *Daemon, container *container.Container) (*types.HealthcheckResult, error) {
+	httpSlice := strslice.StrSlice(container.Config.Healthcheck.Test)[1:]
+
+	endpoint := "/"
+	if len(httpSlice) > 0 {
+		endpoint = httpSlice[0]
+	}
+
+	address := "127.0.0.1"
+
+	// Save the current network namespace
+	origns, err := netns.Get()
+	if err != nil {
+		return nil, err
+	}
+	defer origns.Close()
+
+	ns, err := netns.GetFromPath(container.NetworkSettings.SandboxKey)
+	if err != nil {
+		return nil, err
+	}
+	defer ns.Close()
+
+	defer netns.Set(origns)
+	netns.Set(ns)
```

Comment:
I don't know much about namespaces, but how does this work when there are multiple healthchecks (and other things) going on in the same process?